internal/clients/ec2/ec2_client.go

```diff
@@ -102,7 +102,7 @@ func (c *Client) CreateEC2ClientFromConfig(crd *stsTypes.Credentials) (*Client,
 	return newClient, nil
 }
 
-func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
+func (c *Client) ListInstanceTypes() ([]string, error) {
```

Comment:
I don't think this is sufficient, we will need more info about the instances (memory, cpus...)
We do not do this ATM, but we should do that soon, for the list to be more user readable, so I'd like us to keep returning the full info, or at least do not write it in a way we can't return more info... so maybe custom type?

Reply:
Good point, we are talking with @adiabramovitch actually about the "supported instance types" problem and it too turns out that it would be a great benefit if we had a common type across clouds so the filtering component could be generic. Let's look into that, because I see a big value there in the future when multicloud is a thing. A customer might want need to launch an instance across clouds that is reasonably similar spec. A common type could actually help with this, we should be able to make a function that would take instance type of one cloud and find the closest matches in regard to specs for all other clouds.

This should be possible at least on the technical level (architecture, memory, CPUs, instance storage speed and size, network speed). User is then presented with options and can do the final selection according a different criteria (price, support level etc).

Reply:
So we had a chat with @adiabramovitch and we agreed on creating a new type:

type InstanceType struct {
 Name string,
 CPUs integer,
 MemoryBytes integer, // is perhaps better than MemoryGiB float because dealing with floats is less human friendly
}